src/p5.dance.js

```diff
@@ -247,7 +290,14 @@ module.exports = class DanceParty {
     }
   }
 
-  play(songData, callback) {
+  async play(songData, callback) {
```

Comment:
Seems like an antipattern for `play()` to both take a callback and return a promise.

Reply:
Agreed, but for now they complete at different times. This comes down to whether we care about allowing the dancers to appear before the song starts playing. Previously, that was important to some people. The promise fires immediately if you already have your sprite sheets ready. The callback is to signal the song starting.

And it is partially a case of migrating to the new contract. But, I’d also be ok with removing the callback and switching to a promise only. But the start of rendering will require that everything has loaded and the song is starting. So this is actually more of a change from the current behavior.

Reply:
Resolved by eliminating the `Promise` from `play()` - see below.